### PR TITLE
CXX-788 Fix some -Wall warnings under clang and gcc

### PR DESCRIPTION
**Pull request by acmorrow**
Fixes the warnings identified in CXX-788 and several others.

### CODE REVIEW

File: src/bsoncxx/private/error_category.cpp [hunk 1]
```diff
@@ -120,7 +122,7 @@ class libbson_unknown_error final : public std::error_category {
         return "libbson unknown error";
     }
 
-    std::string message(int code) const noexcept override {
```
Comment (samantharitter) [inline src/bsoncxx/private/error_category.cpp]: Was having an unused named parameter giving us an unused variable warning?

File: src/mongocxx/exception/error_code.hpp [hunk 2]
```diff
@@ -25,6 +25,7 @@ enum class error_code : std::int32_t {
 	k_invalid_client_object = 1,
 	k_invalid_collection_object,
 	k_invalid_database_object,
+        k_invalid_parameter,
```
Comment (samantharitter) [inline src/mongocxx/exception/error_code.hpp]: extra space?

Comment (samantharitter) [inline src/mongocxx/exception/error_code.hpp]: Also, can you add a case for `k_invalid_parameter` to `mongocxx_error_category_impl::message()` ?

File: src/bsoncxx/private/error_category.cpp (hunk 1 above)
Comment (acmorrow) [inline src/bsoncxx/private/error_category.cpp]: Yes, with -Wextra. It is pretty pedantic, but it is fair that you declared a name for a parameter but then didn't use it.

File: src/mongocxx/exception/error_code.hpp (hunk 2 above)
Comment (acmorrow) [inline src/mongocxx/exception/error_code.hpp]: The weird spacing is an artifact - apparently this list is using leading tabs? I've cleaned it up, but we can leave the real fix for clang-format after rc0

Comment (acmorrow) [inline src/mongocxx/exception/error_code.hpp]: Added to the message enum, good catch, thanks.

**Comment by acmorrow** [comment]
Thanks @samantharitter, updated. PTAL.

File: src/bsoncxx/private/error_category.cpp (hunk 1 above)
Comment (samantharitter) [inline src/bsoncxx/private/error_category.cpp]: Makes sense, thanks

**Comment by samantharitter** [comment]
LGTM!

**Pull request merged**